### PR TITLE
config improvements

### PR DESCRIPTION
**Pull request by fmigneault**
Before this fix, the pasted ini file would define most settings, but ini file employed by db migration would use the default `./config/magpie.ini` path.

We allow also json/yaml file extensions as these formats where already supported for permissions.cfg or providers.cfg, but not explicitly looked for when specifying a directory path to load many files.

# commits 
- employ gunicorn paste ini file for db migration
- yml/json configs extensions support

### CODE REVIEW

**Comment by fmigneault** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- magpie/app.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fmigneault/Magpie/pullRequest?prid=5130814&bid=16860872)

File: magpie/helpers/run_db_migration.py [hunk 1]
```diff
@@ -1,4 +1,13 @@
...
 
 if __name__ == "__main__":
-    run_database_migration()
+    parser = argparse.ArgumentParser(description="Run Magpie database migration")
+    parser.add_argument("-c", "--config-file", metavar="config_file", dest="config_file", type=str,
+                        default=MAGPIE_INI_FILE_PATH,
+                        help="configuration file to employ for database connection settings "
+                             "(default: MAGPIE_INI_FILE_PATH='%(default)s)'")
+    args = parser.parse_args()
+    run_database_migration(settings={"magpie.alembic_ini_file_path": args.config_file})
```
Comment (davidcaron) [inline magpie/helpers/run_db_migration.py]:
```suggestion
    run_database_migration(settings={"magpie.ini_file_path": args.config_file})
```

I could be wrong, but I think it needs to be `magpie.ini_file_path` here.

Comment (fmigneault) [reply magpie/helpers/run_db_migration.py]: Yes you're right!

**Comment by fmigneault** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- magpie/app.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fmigneault/Magpie/pullRequest?prid=5130814&bid=16860872)

**Comment by fmigneault** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- magpie/app.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fmigneault/Magpie/pullRequest?prid=5130814&bid=16860872)

**Comment by fmigneault** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- magpie/app.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fmigneault/Magpie/pullRequest?prid=5130814&bid=16860872)

**Pull request merged**